example/App.js

```diff
@@ -16,45 +21,113 @@ export default class AppContainer extends React.Component {
+        <Text style={{ fontSize: 15, color: '#fff' }}>{title}</Text>
+      </Entypo.Button>
+    </View>
+  );
+
+  _renderSelectionText = () => {
+    const { selectedIndex } = this.state;
+    const text = selectedIndex === null
+      ? 'No Option Selected'
+      : `Option #${selectedIndex + 1} Selected`;
+
+    return (
+      <View style={{ marginTop: 20 }}>
+        <Text style={{ textAlign: 'center', color: 'blue', fontSize: 16 }}>
+          {text}
+        </Text>
+      </View>
+    )
+  };
+
   render() {
     return (
-      <View style={styles.container}>
-        <View style={{ marginBottom: 30 }}>
-          <Text style={{ textAlign: 'center' }}>
-            Hello! This is a simple example app to demonstrate @expo/react-native-action-sheet.
+      <View style={{ flex: 1 }}>
+        <ScrollView style={styles.scrollView} contentContainerStyle={styles.contentContainer}>
+          <View style={{ marginBottom: 30 }}>
+            <Text style={{ textAlign: 'center', fontSize: 16 }}>
+              {"Hello!\n\nThis is a simple example app to demonstrate @expo/react-native-action-sheet."}
+            </Text>
+          </View>
+          {this._renderButton('Options Only', this._onOpenActionSheet(false, false, false, false, false))}
+          {this._renderButton('Title', this._onOpenActionSheet(true, false, false, false, false))}
+          {this._renderButton('Title & Message', this._onOpenActionSheet(true, true, false, false, false))}
+          {this._renderButton('Icons', this._onOpenActionSheet(false, false, true, false, false))}
+          {this._renderButton('Title, Message, & Icons', this._onOpenActionSheet(true, true, true, false, false))}
+          {this._renderButton('Use Separators', this._onOpenActionSheet(true, false, true, true, false))}
+          {this._renderButton('Custom Styles', this._onOpenActionSheet(true, true, true, false, true))}
+          {this._renderSelectionText()}
+          <Text style={{ marginTop: 32 }}>
+            Note: Icons and custom text styles are only available on Android. Separators can only be toggled on Android; they always show on iOS.
           </Text>
-        </View>
-
-        <Entypo.Button name="code" backgroundColor="#3e3e3e" onPress={this._onOpenActionSheet}>
-          <Text style={{ fontSize: 15, color: '#fff' }}>Open action sheet</Text>
-        </Entypo.Button>
+        </ScrollView>
       </View>
     );
   }
 
-  _onOpenActionSheet = () => {
+  _onOpenActionSheet = (
+    withTitle: boolean,
+    withMessage: boolean,
+    withIcons: boolean,
+    withSeparators: boolean,
+    withCustomStyles: boolean,
+  ) => () => {
     // Same interface as https://facebook.github.io/react-native/docs/actionsheetios.html
-    let options = ['Delete', 'Save', 'Cancel'];
-    let destructiveButtonIndex = 0;
-    let cancelButtonIndex = 2;
+    const options = ['Delete', 'Save', 'Share', 'Cancel'];
+    const icons = withIcons ? [deleteIcon, saveIcon, shareIcon, cancelIcon] : null;
```

Comment:
re: icons - it would be great to utilize icons already shipped in the project

Reply:
For sure! I haven’t used expo so I didn’t realize it had icons already. 🎉